src/3d/qgs3dmapsettings.cpp

```diff
@@ -313,6 +314,7 @@ void Qgs3DMapSettings::readXml( const QDomElement &elem, const QgsReadWriteConte
   mShowCameraRotationCenter = elemDebug.attribute( QStringLiteral( "camera-rotation-center" ), QStringLiteral( "0" ) ).toInt();
   mShowLightSources = elemDebug.attribute( QStringLiteral( "show-light-sources" ), QStringLiteral( "0" ) ).toInt();
   mIsFpsCounterEnabled = elemDebug.attribute( QStringLiteral( "show-fps-counter" ), QStringLiteral( "0" ) ).toInt();
+  mIsDebugOverlayEnabled = elemDebug.attribute( QStringLiteral( "show-debug-overlay" ), QStringLiteral( "0" ) ).toInt();
```

Comment:
I don't think this is necessary to persist -- it would work better as a transient setting which is discarded when the project is closed

Reply:
Indeed, this makes sense.